src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Refactoring/RenameEndpoint.cs

```diff
@@ -88,6 +100,46 @@ internal class RenameEndpoint : IRenameEndpoint
                 return null;
             }
 
+            var razorEdits = await TryGetRazorComponentRenameEditsAsync(request, requestDocumentSnapshot, codeDocument, cancellationToken).ConfigureAwait(false);
+
+            if (razorEdits is not null)
+            {
+                return razorEdits;
+            }
+
+            // If we're not doing single server rename, then we're done. C# and Html will be handled by the RenameHandler in the HtmlCSharp server.
+            if (!_languageServerFeatureOptions.SingleServerRenameSupport)
+            {
+                return null;
+            }
+
+            var sourceText = await documentContext.GetSourceTextAsync(cancellationToken).ConfigureAwait(false);
+            if (!request.Position.TryGetAbsoluteIndex(sourceText, _logger, out var absoluteIndex))
```

Comment:
Do we not do something similar in the pre-existing "TryGetRazorComponentRenameEdit" bit? Seems like somewhere around the line we'll need to holistically convert to absolute index. Might be good to pass this info through to both APIs

Reply:
Yeah, can do. I tried to leave the old code untouched, to make reviewing easier, but makes sense to combine these two. I'm also going to move the projection check up, because we don't need to do any work of our own, if we're in a C# context, but again that muddies the diff.

Now that you've all seen the first draft, I'm okay with being a bit more bold :D